### PR TITLE
Update app platform overview page with recent changes

### PR DESCRIPTION
**Pull request by rossf7**
### Please remember to

- [x] Give the PR a meaningful title -- it will be shown in the release notes
- [x] Add (or remove) [user questions](https://github.com/giantswarm/docs/blob/master/CONTRIBUTING.md#front-matter) associated with any updated docs

### CODE REVIEW

File: src/content/app-platform/overview/index.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ you to browse, install and manage the configurations of apps (such as Prometheus
 from a single place; the management cluster.
 
 Using this platform, we are providing a collection of curated _Apps_. These _Apps_ are grouped into _App Catalogs_, which are browsable through our web interface.
+We also use app platform to install the apps that are pre-installed in your cluster (such as CoreDNS).
```
Comment (rossf7) [inline src/content/app-platform/overview/index.md:34]: Adding this to make it clear that app platform is used even if the customer is not using managed apps.
See https://github.com/giantswarm/giantswarm/issues/16903

@Oshratn PTAL 🙏

File: src/content/app-platform/overview/index.md [hunk 2]
```diff
@@ -111,14 +112,13 @@ It’s possible to create your own App Catalog. This is useful if you want to cr
 
 ### How can I interact with the Giant Swarm App Platform
 
-You can interact with the Giant Swarm App Platform through our API and
-our web interface.
+You can interact with the Giant Swarm App Platform through creating App custom resources using the Management API, our web interface, and our REST API.
```
[the remark this answers is not in the document]
Comment (marians) [reply src/content/app-platform/overview/index.md:115]: > Management API, our web interface, and our REST API

These could be linked ideally (unless that already happened above on the same page. Haven't checked.)

[the remark this answers is not in the document]
Comment (marians) [reply src/content/app-platform/overview/index.md:115]: Well, now I notice that the links are coming underneath. Never mind.

**Review by marians — APPROVED** [review]

**Review by Oshratn — APPROVED** [review]
LGTM 🚀

**Pull request merged**